### PR TITLE
refactor(disk): Refactor sparse disk to sparse block device

### PR DESCRIPTION
**Pull request by akhilerm**
Sparse disk is refactored  to block device resource. Only block device resources will be created for sparse files. These block devices of type sparse can be used for creating sparse pool.

Fixes https://github.com/openebs/node-disk-manager/issues/250

### CODE REVIEW

File: cmd/ndm_daemonset/controller/diskstore.go [hunk 1]
```diff
@@ -162,14 +162,13 @@ func (c *Controller) GetExistingDiskResource(listDr *apis.DiskList, uuid string)
...
 func (c *Controller) DeactivateStaleDiskResource(devices []string) {
-	listDisks := append(devices, GetActiveSparseDisksUuids(c.HostName)...)
 	listDR, err := c.ListDiskResource()
 	if err != nil {
 		glog.Error(err)
 		return
 	}
 	for _, item := range listDR.Items {
-		if !util.Contains(listDisks, item.ObjectMeta.Name) {
+		if !util.Contains(devices, item.ObjectMeta.Name) {
```
Comment (imazik) [inline cmd/ndm_daemonset/controller/diskstore.go]: disk and device confuse me here.
if devices does not have a device then deactivate disk.
should we change `DeactivateDisk`
cc @akhilerm .

File: cmd/ndm_daemonset/controller/sparsefilegenerator.go [hunk 2]
```diff
@@ -68,10 +68,10 @@ const (
 	//SparseFileDefaultCount defines the default sparse count files
 	SparseFileDefaultCount = "1"
 
-	//SparseDiskType defines sparse disk type
-	SparseDiskType = "sparse"
-	//SparseDiskPrefix defines the prefix for the sparse disk
-	SparseDiskPrefix = "sparse-"
+	//SparseBlockDeviceType defines sparse disk type
```
Comment (imazik) [inline cmd/ndm_daemonset/controller/sparsefilegenerator.go]: `sparse disk` -> `sparse file` ?

Comment (imazik) [reply cmd/ndm_daemonset/controller/sparsefilegenerator.go]: or `sparse device`

File: cmd/ndm_daemonset/controller/diskstore.go (hunk 1 above)
Comment (akhilerm) [reply cmd/ndm_daemonset/controller/diskstore.go]: @imazik  The following should fix the ambiguity.
```
if !util.Contains(disks, item.ObjectMeta.Name) {
```

> should we change `DeactivateDisk`

 You meant `DeactivateStaleDiskResource` ?

File: cmd/ndm_daemonset/controller/sparsefilegenerator.go (hunk 2 above)
Comment (akhilerm) [reply cmd/ndm_daemonset/controller/sparsefilegenerator.go]: `sparse device` sounds good to me

**Pull request merged**